src/metpy/calc/tools.py

```diff
@@ -1175,13 +1418,16 @@ def gradient(f, axes=None, coordinates=None, deltas=None):
 
     See Also
     --------
-    laplacian, first_derivative
+    laplacian, first_derivative, vector_derivative
 
     Notes
     -----
     If this function is used without the `axes` parameter, the length of `coordinates` or
     `deltas` (as applicable) should match the number of dimensions of `f`.
 
+    This will not give projection-correct results when applied to the components of a vector
+    quantity. Instead, for vector quantities, use `vector_derivative`.
```

Comment:
```suggestion
    This will not give projection-correct results for horizontal geospatial fields. Instead,
    for vector quantities, use `vector_derivative`, and for scalar quantities, use
    `geospatial_gradient`.
```